spec/timezone.html

```diff
@@ -5,6 +5,97 @@
+      When called with any other string, it will throw a *RangeError* exception.
+    </p>
+    <p>
+      Implementations are encouraged to support the Zone and Link names of the IANA Time Zone Database, and to use the best available current and historical information about their offsets from UTC and their daylight saving time rules in calculations.
+    </p>
+    <p>
+      An ECMAScript implementation that includes the ECMA-402 Internationalization API has additional requirements for the built-in time zones that must be supported, as specified in the ECMA-402 specification.
+    </p>
+
+    <emu-note type="editor">
+      <p>These host hooks need to be added to ES2021 <emu-xref href="#sec-host-hooks-summary"></emu-xref> when merged in.</p>
+    </emu-note>
+
+    <emu-clause id="sec-hostisvalidtimezonename" aoid="HostIsValidTimeZoneName">
+      <h1>HostIsValidTimeZoneName ( _timeZone_ )</h1>
+
+      <p>
+        HostIsValidTimeZoneName is a host-defined abstract operation that takes argument _timeZone_ (a String value).
+        It verifies that _timeZone_ represents a valid built-in time zone.
+      </p>
+      <p>
+        The implementation of HostIsValidTimeZoneName must conform to the following requirements:
+      </p>
+      <ul>
+        <li>It must always complete normally (i.e., not return an abrupt completion.)</li>
+        <li>It must always return either *true* or *false*.</li>
+      </ul>
+      <p>
+        The implementation of HostIsValidTimeZoneName should return *true* if _timeZone_, converted to upper case, is equal to one of the built-in time zone names, converted to upper case.
+        It should return *false* otherwise.
+      </p>
+      <p>The minimum implementation of HostIsValidTimeZoneName, supporting only the *"UTC"* time zone, performs the following steps when called:
+      </p>
+
+      <emu-alg>
+        1. Assert: Type(_timeZone_) is String.
+        1. Let _tzText_ be ! StringToCodePoints(_timeZone_).
+        1. Let _tzUpperText_ be the result of toUppercase(_tzText_), according to the Unicode Default Case Conversion algorithm.
+        1. Let _tzUpper_ be ! CodePointsToString(_tzUpperText_).
+        1. If _tzUpper_ and *"UTC"* are the same sequence of code points, return *true*.
+        1. Return *false*.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-hostcanonicalizetimezonename" aoid="HostCanonicalizeTimeZoneName">
+      <h1>HostCanonicalizeTimeZoneName ( _timeZone_ )</h1>
+
+      <p>
+        HostCanonicalizeTimeZoneName is a host-defined abstract operation that takes argument _timeZone_ (a String value).
+        It returns the canonical and case-regularized form of the _timeZone_ argument.
+      </p>
+      <p>
+        The _timeZone_ parameter must be a valid time zone name according to HostIsValidTimeZoneName.
```

Comment:
Is this meant to be an assertion, meaning the spec is responsible for ensuring that this host hook is only ever called with a valid time zone name? (the below minimum impl implies it is)

Reply:
Yes, that's correct, I've revised the language a bit, hopefully it's clear now.